src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1627,6 +1625,46 @@ public void MainWindowTitle_GetWithGui_ShouldRefresh_Windows()
             }
         }
 
+        [PlatformSpecific(TestPlatforms.Windows)] // it tests Windows implementation
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void RespondingIsRefreshedAfterEveryCallToRefresh()
```

Comment:
This just tests _haveResponding. Refresh() clears many things not just _haveResponding. Eg., _mainWindowTitle which was also recently fixed, plus _processInfo, etc.

An alternative approach to test Refresh() could possibly be, to set all the private fields that Refresh() clears, then call it and verify they were cleared? That doesn't involve launching any process, and isn't racy.

Reply:
good idea! #done